### ISSUE TITLE
[QUESTION] Privacy of Qwant and Startpage

### ISSUE THREAD

**Issue by lilithium-hydride** [Lissy93/awesome-privacy#45]
### Your Question

I had never really looked into Qwant or Startpage before and was curious to see if their claims of privacy actually held up (spoiler alert: it doesn't look like it).

For Qwant, I went to the homepage, searched for "test", and then saved uBlock Origin's blocked requests, which appear in the table below. Note that as you stay on the page and interact with random elements, further requests will be sent and blocked. I don't know what some of these requests are for, and I can assume some of them are extraneous without being necessarily malicious, but `rum` stands for Real User Monitoring, and is a tracker capable of collecting a whole host of information on the user. What exactly is Qwant collecting with RUM? What does it do with this data? There's no way to be sure, because Qwant isn't open source and the payloads sent to the `apm/intake/v2/rum/events` endpoint are garbled binary data.

<details><summary>Logger output</summary>

| | | | |
|:--- |:--- |:--- |:--- |
| /rum/events | `www.qwant.com` | xhr | `https://www.qwant.com/apm/intake/v2/rum/events` |
| /rum/events | `www.qwant.com` | xhr | `https://www.qwant.com/apm/intake/v2/rum/events` |
| \|\|qwant.com/action/ | `www.qwant.com` | ping | `https://www.qwant.com/action/display` |
| \|\|qwant.com/action/ | `www.qwant.com` | ping | `https://www.qwant.com/action/display` |
| ##.result--ext | `www.qwant.com` | dom | `https://www.qwant.com/?q=test&t=web` |
| \|\|qwant.com/action/ | `www.qwant.com` | ping | `https://www.qwant.com/action/display` |
| \|\|qwant.com/action/ | `www.qwant.com` | ping | `https://www.qwant.com/action/display` |
| \|\|qwant.com/action/ | `www.qwant.com` | ping | `https://www.qwant.com/action/display` |
| \|\|qwant.com/action/ | `www.qwant.com` | ping | `https://www.qwant.com/action/display` |
| \|\|qwant.com/action/ | `www.qwant.com` | ping | `https://www.qwant.com/action/display` |
| \|\|qwant.com/action/ | `www.qwant.com` | ping | `https://www.qwant.com/action/display` |
| \|\|qwant.com/action/ | `www.qwant.com` | ping | `https://www.qwant.com/action/display` |
| \|\|qwant.com/action/ | `www.qwant.com` | ping | `https://www.qwant.com/action/display` |
| \|\|qwant.com/action/ | `www.qwant.com` | ping | `https://www.qwant.com/action/ui` |
| /rum/events | `www.qwant.com` | xhr | `https://www.qwant.com/apm/intake/v2/rum/events` |
| /rum/events | `www.qwant.com` | xhr | `https://www.qwant.com/apm/intake/v2/rum/events` |
| /rum/events | `www.qwant.com` | xhr | `https://www.qwant.com/apm/intake/v2/rum/events` |
| /rum/events | `www.qwant.com` | xhr | `https://www.qwant.com/apm/intake/v2/rum/events` |
| \|\|qwant.com/action/ | `www.qwant.com` | ping | `https://www.qwant.com/action/display` |
| \|\|qwant.com/action/ | `www.qwant.com` | ping | `https://www.qwant.com/action/ui` |
| /rum/events | `www.qwant.com` | xhr | `https://www.qwant.com/apm/intake/v2/rum/events` |
| /rum/events | `www.qwant.com` | xhr | `https://www.qwant.com/apm/intake/v2/rum/events` |
| \|\|qwant.com/v2/api/ux/surveys? | `www.qwant.com` | xhr | `https://api.qwant.com/v2/api/ux/surveys?website=qwant&tab=home&tgp=90&locale=en_US&device=desktop` |
| \|\|qwant.com/action/ | `www.qwant.com` | ping | `https://www.qwant.com/action/webapp_loaded` |
| \|\|qwant.com/action/ | `www.qwant.com` | ping | `https://www.qwant.com/action/display` |
| \|\|qwant.com/action/ | `www.qwant.com` | ping | `https://www.qwant.com/action/display_page` |
</details>

As for Startpage, it doesn't send near as many shady-looking requests, but I also wouldn't call it private. I won't bog you down with another uBlock Origin log, as [this GitHub issue](https://github.com/uBlockOrigin/uAssets/issues/13141) does a good job of summing it up.

Should these still be included? They're certainly better than `$BIG_TECH_SEARCH`, but I'm also not sure if they're the best places to direct people. 

### Please tick the boxes

- [X] You have filled out this form accurately, and to the best of your knowledge
- [X] A similar question has not already [been asked](https://github.com/Lissy93/awesome-privacy/issues?q=is%3Aissue) for this software/ service
- [X] You agree to the [code of conduct](https://github.com/Lissy93/awesome-privacy/blob/main/.github/CODE_OF_CONDUCT.md)

**Comment by Lissy93** [comment]
Looking at the web request you mentioned. It is indeed analytics...

It is going to their self-hosted instance of Elastic (see [docs](https://www.elastic.co/guide/en/apm/guide/current/api-events.html)).

<details>
  <summary><b>For reference, here's the full request</b></summary>


#### Target:
```
POST  https://www.qwant.com/apm/intake/v2/rum/events
```

#### Body:
```json
{"metadata":{"service":{"name":"phoenix","version":"cb73d8fab484b85a110fd3129702240dcc3ede59","agent":{"name":"rum-js","version":"5.12.0"},"language":{"name":"javascript"}}}}
{"transaction":{"id":"b2501a075c81f532","trace_id":"d2d526bfc5fe54d7ef1e2f8f3bf37caf","name":"Click - div","type":"user-interaction","duration":3108,"context":{"custom":{"classes":"Flex-module__flex___35zId"},"page":{"referer":"","url":"https://www.qwant.com/?q=test&t=web"}},"span_count":{"started":0},"sampled":false,"sample_rate":0}}
```

#### Request Headers:
```
Host: www.qwant.com
User-Agent: Mozilla/5.0 (X11; Linux x86_64; rv:101.0) Gecko/20100101 Firefox/101.0
Accept: */*
Accept-Language: en-GB,en;q=0.5
Accept-Encoding: gzip, deflate, br
Referer: https://www.qwant.com/
Content-Type: application/x-ndjson
Content-Length: 512
Origin: https://www.qwant.com
Connection: keep-alive
Cookie: hide_addon_dialog=1; theme=1; s=2; cookieAlertSettings=1
Sec-Fetch-Dest: empty
Sec-Fetch-Mode: cors
Sec-Fetch-Site: same-origin
DNT: 1
Sec-GPC: 1
```

#### Server Headers:
```
content-security-policy | script-src  'self' 'unsafe-inline' 'unsafe-eval' data: *.qwant.com qwant.com;  style-src 'unsafe-inline' data: *.qwant.com *.qwantjunior.com qwant.com;  base-uri 'self'; block-all-mixed-content; frame-ancestors *.qwant.com  *.qwantjunior.com lmqt.fyi; worker-src blob: 'self'  www.youtube-nocookie.com www.youtube.com; object-src 'self'; img-src  blob: 'self' s1.qwant.com s2.qwant.com s.qwant.com f.qwant.com data:  s-lite.qwant.com www.qwant.com; form-action 'self'; script-src-elem  'self' 'unsafe-inline' 'unsafe-eval' data: *.qwant.com qwant.com;  default-src 'self' data: blob:; connect-src *.qobuz.com *.apple.com  *.qwant.com qwant.com extras.qwantjunior.com; font-src 'self'; media-src  blob: *.qwant.com *.apple.com *.qobuz.com *.vid.web.acsta.net;  frame-src viewer.dood3d.com *.vid.web.acsta.net player.twitch.tv  player.vimeo.com www.dailymotion.com players-cdn.vidmizer.com  players-cdn-v2.vidmizer.com *.qwant.com *.qwantjunior.com  www.youtube-nocookie.com *.tvlocale.fr *.smartrezo.com  *.femmesetcitoyennete.fr *.jeunesreporterssansfrontieres.fr  *.medias-francophones.com *.trendy-community.fr *.tvcitoyenne.com  *.veitech.com *.localetv.eu player.myvideoplace.tv;
content-type | application/json
date | Tue, 26 Jul 2022 14:22:30 GMT
permissions-policy | speaker=(self),  vibrate=(), oversized-images=(), gyroscope=(), magnetometer=(),  document-domain=(), interest-cohort=(), microphone=(),  display-capture=(), encrypted-media=(), midi=(), geolocation=(self),  accelerometer=(), autoplay=(self), payment=(), vr=(), fullscreen=(),  camera=(), ambient-light-sensor=(), sync-xhr=(self  "https://api.qwant.com"), usb=()
referrer-policy | origin
server | Qwant
strict-transport-security | max-age=31536000 ; includeSubdomains ; preload
x-content-type-options | nosniff
x-frame-options | SAMEORIGIN
x-xss-protection | 1; mode=block
```

#### Response:
```
None
```
</details>


^^ There's other requests too, but they all seem to follow this format.

---

I started looking into Qwant a bit further.... 


### Privacy Policy
Started with their [Privacy Policy](https://about.qwant.com/en/legal/confidentialite/). It's really hard to read as info is scattered around, which IMO is not a good sign. Some things that jumped out at me:

- They collect analytics
  - including user agent, referer (where u came from), the meta info about the links you clicked (unclear if this includes the actual URL), and IP address
- They could be sharing data with third-parties
  - _"We pseudonymize what we need to keep for statistics and for transfer to our technology and business partners."  _
- They use third-party services, but are not transparent about what for and when
  - But looking at the sources, Deepl for language translations, Bing for ads,

I couldn't find it written, but search queries have to be somehow sent to Bing in order to serve up ads. This is happening server-side, so there's no reason any other user info would be forwarded, but again, not clear.

### Tracking

Qwant is using a self-hosted instance of, I think Elastic APM for both user analytics and "performance monitoring" (actually just more user analytics)

I tried using it as a user would, and analysing the client-side requests that were being sent. And I could not see anything other than what was shown above, which was particularly alarming.

They've [previously used](https://user-images.githubusercontent.com/1862727/181016104-5493c916-5c53-4d2d-a52c-30c5f6185b4e.png) New Relic, TraderDoubler and Google Analytics (wtf?!), for analytics and user and performance tracking. That's based on historical [BuiltWith scans](https://builtwith.com/detailed/qwant.com), so I cannot properly verify that it's correct myself. I did check [old snapshots](https://web.archive.org/web/20140101000000*/https://www.qwant.com) on archive.org, and it seems to add up. 

### Cookies
There use of local storage, session storage, and cookies looks all fine. Each setting is stored locally, as opposed to giving you a user ID and storing settings remotely. There's no 3rd-party cookies. I played around with it for a while, and [this](https://user-images.githubusercontent.com/1862727/181011839-d79e3c52-93ce-4b99-b7fd-2c55a3d6e72e.png) was all that was stored - so just my user settings. They have a 6-month expiry.


### JS
If you disable JavaScript, you're redirected to https://lite.qwant.com/ - which functions entirely on the server-side, and IMO is actually a much better experience. There's also no client-side analytics.

### Ads

The ads are provided by Bing/ Microsoft. That does mean the search term is sent to Mircosoft, although it happens server-side, so there shouldn't be any user data attached. They're contextual so based only on the current search, and not user data. 

One thing I will say, is that the search result ads are labelled as ads, whereas the suggested products, and shopping results have affiliate links attached, yet are not labelled as adverts. Clicking ads is pretty bad for privacy, so it should be important that they're properly labelled.

I think the 2 issues here are 1) Microsoft and 2) not being clear when your clicking an ad.
But just having ads alone I don't think is an issue, as it would be more of a red flag is they didn't have a clear revenue model.

### Security
I couldn't find any published security audit.
Their privacy policy barley mentions security, or the practical things they've put in place to maintain integrity and protect user data. 

But in terms of the very basic easy to check stuff, like [headers](https://securityheaders.com/?followRedirects=on&hide=on&q=www.qwant.com), [certificates](https://www.ssllabs.com/ssltest/analyze.html?d=qwant.com&latest), [ports](https://dnschecker.org/port-scanner.php?query=194.187.168.100&ptype=server) everything looks normal.

You're not trusting them with a lot personal info, so I guess the biggest motivation of an adversary would be to manipulate results.

### Criticism

Overall, I couldn't find a lot of complaints, no historical big privacy mishaps, data leaks or security concerns. The [Criticism Section of their Wikipedia](https://en.wikipedia.org/wiki/Qwant#Criticism) doesn't list anything of great concern. There's not many compleints on [?q=Qwant on r/privacy](https://www.reddit.com/r/privacy/search/?q=Qwant&restrict_sr=1&sr_nsfw=)


### Search

Not sure if it's relevant, but unlike a lot of other _"search engines"_ Qwant does do their own crawling and indexing, as [explained in this article](https://medium.com/qwant-blog/web-indexation-where-does-qwants-independence-stand-8eab4f7856f8).

They also appear to own their own infrastructure, and are self-hosting most of the services they use.

Worth noting that they are popular in France and French-speaking countries, but less so in other regions.

---

TLDR; user analytics, association with Microsoft, the use of redirect URLs, Bing ads, somewhat unclear privacy policy.

I'm pretty neutral on this. There's definitely some issues, but weather the severity is enough to warrant removal, I'm not sure.
But I will look into it further, and definitely update the listing to mention them.  And if anyone else has anything to add, please do :)

If anyone else has any thoughts on this, or if I've missed anything, or got something wrong, do let me know. And if there is stuff I've missed then I'm happy to consider removing Qwant, but it's a slippery slope, as when you look into anything hard enough, there's always going to be issues. I think the main thing is just to present the facts, and let users decide what's right for them.

**Comment by Lissy93** [comment]
I'll take a look at StartPage also, when I get a moment. StartPage has had a lot more controversy over the years, so there should be more info to go on.

DuckDuckGo also has some issues, so I should re-check that too.

Side note: I think a common issue that people have when trying to switch to a more privacy-respecting search engine, is just that the results either aren't as tailored as what most people have gotten used to. Which in turn means they end up going back to big 'G. So quality of search results is probably also a factor that should be taken into account.

**Comment by lilithium-hydride** [comment]
Wow, that's a lot of good research! 

> I'm happy to consider removing Qwant, but it's a slippery slope, as when you look into anything hard enough, there's always going to be issues. I think the main thing is just to present the facts, and let users decide what's right for them.

Yeah, I agree. You can dig and find unsavory behavior with a *lot* of things on this list, and staying fully informed to make educated decisions is the real goal with this sort of thing. I think it would do a lot of good to put all of your findings into another document somewhere and link to it so that it doesn't just get buried somewhere in an issue tracker. 

I was tempted to mention DuckDuckGo as they've always weirded me out, but I figured that horse had been resurrected and beaten to death many times throughout its existence and I probably wouldn't present any new information. I'm curious to see what you find, though.

**Comment by atomGit** [comment]
yeah i know, and older topic, but i wanted to splice in a few comments...

@Lissy93 said...

> If you disable JavaScript, you're redirected to https://lite.qwant.com/ 

testing this now in 2023 and the forwarding doesn't occur

also i thought JS was necessary to get Qwant to display anything, but either i was wrong or they changed something

> I'll take a look at StartPage also, when I get a moment. StartPage has had a lot more controversy over the years, so there should be more info to go on.
> 
> DuckDuckGo also has some issues, so I should re-check that too.

i've been lookin' more closely at search engines [since 2017](https://12bytes.org/articles/tech/alternative-search-engines-that-respect-your-privacy/) and the conclusions i've come to are simple: if they don't crawl the web themselves, ignore it - Qwant (Bing), DDG (Bing), Startpage (Google), etc. are all **crap** from a privacy POV and from a results POV all the rest of the proxies are garbage

the remainder that do indexing, and have decent privacy policies, and that don't require JS, is a _very_ small number with [Mojeek](https://www.mojeek.com/about/) probably being the best of them at the moment (around 6.5 billion pages indexed as of this writing)

**Comment by ltguillaume** [comment]
@lissy93 What's your verdict on Qwant several years after https://github.com/lissy93/awesome-privacy/issues/45#issuecomment-1195606297? I know it's listed, but I wonder if that means it's all good now 😉

**Comment by lissy93** [comment]
I'll re-open this. At a minimum, the current listing is out-dated (says it doesn't use cookies, it does). I'll look into it properly some point this week, and the other search engines too.

> I know it's listed, but I wonder if that means it's all good now 😉

Nothing is ever all good 😅🥹

**Comment by lissy93** [comment]
Ok... here's what I've found


### Summary

- 🟢 No ad cookies, and ads are contextual (to the resutls), instead of being tied to you as a user
- 🟢 All image results get proxied, so as to never expose visitor IP to third-parties
- 🟢 Overall fair security basics (except few small header issues noted below), plus a security.txt and bounty program
- 🟢 (apparently) no selling of user data to anyone (cannot verify)
- 🟢 Working (with Ecosia) on the European Search Perspective to build their own, independent search index, free from Bing
- 🟡 Still uses results from Bing, plus Bing Ads. But all apparently properly proxied and no user info shared with MS (cannot prove/disprove, but this is likely correct)
- 🟡 French company + hosted in France. Part of the 9-eyes
- 🟡 Mobile app has 1 tracker (Mozilla telemetry) and many permissions
- 🟠 Turbulent leadership, and apparently in financial trouble
- 🟠 Cookies (they didn't used to), but now have semi x-site tracking cookies
- 🟠 Uses DataDome for bot protection (likely in response to bot abuse. Understandable, but not good for privacy)
- 🔴 Major inconsistences in privacy policy (see below)

---

### Privacy Policy out-dated and incorrect
I started with the privacy policy (the English version, at: https://about.qwant.com/en/legal/confidentialite). Immediately noted DataDome is missing, despite it being visible on all requests.

Turns out the English (and German, Spanish and Italian) version hasn't been updated since 2024, and is missing a lot of stuff which appears present in the French version (2026).

For context, DataDome is an anti-bot platform, in order to detect bots it collect/analyses usage data, like mouse movements, scrolls, clicks, etc. It's not great for privacy, since they receive a lot of user data. So it really should be mentioned in their policy. You can verify yourself that DataDome IS present, just try opening in incognito / or when on VPN (in the UK, despite it missing from English policy).

Also a lot of other things seem to be missing. Including Mistral AI and Staan.ai - both used for the AI search results, and can be verified with `?llm=true` to enable/force AI answers, even in UK (and Spain/Germany/Italy).

So fair to assume that the rest of this privacy policy is not accurate or complete, and should not be trusted. Not great for a "privacy" company.

---

### Search query IS tracked in analytics

The Qwant homepage proudly displays "Qwant does not retain your search data". Sounds great! But inspecting the requests it seems that this might not actually be the case.

The /action/display_page route sends both my literal search string AND all my tracking identifiers. This beacon gets sent every few seconds, whichever page you're on (including homepage, privacy policy, qwant junior/kids page, SRP, etc).

So, the homepage statement is false. But the statement in the privacy policy initially appears bit more correct, it says browsing data incl. search keywords is collected _"automatically and anonymously"_. But actually, this too is false - if you're tying a user's search query, with their persistent 13-month unique ID plus device identifiers and IP address, then that is NOT anonymous. 

AND this gets shared cross-domain, so that `shdw.me` and `www.qwantjunior.com` can share the tracking love too.

Even though the privacy policy explicitly says _"There is no data transfer between SHADOW and QWANT. SHADOW does not have access to your personal data in any way."_. Furthermore, this cross-domain linkage also disqualifies the CNIL analytics-exemption that Qwant relies on. All 6 of the tracking tags from Piwik container are flagged as `"consentType":"not_require_consent"`

Here's a pretty picture showing the above:

<img width="800" alt="Image" src="https://github.com/user-attachments/assets/5b82c7ac-88b2-4db1-b0f0-460a495c248d" />


---

### No cookie consent banner shown to UK users

The CMP is hard-gated to France, Spain, USA, Austria, Belgium. Germany, Italy, Ireland, Netherlands, Poland. So UK never consent to cookies BUT the cookies are planted anyway. Just because we left the EU doesn't mean we don't have cookie laws here Qwant! GDPR still applies here. Seems there's a similar gap for logged-out French users, although I cannot be certain of this.

Anyway, even though I've not agreed to cookies, Qwant still plants: `_pk_id.xxx` (393 day unique visitor ID), `_pk_ses.xxx` (session-only, tracks session), `datadome` (1 year, a unique device ID) and `ab_test_group` (180 days, A/B bucket).

AND for the rest of users, the banner only shows on the SRP, not initially on the homepage. Qwant says that this is fine, because the only homepage cookies are for _"Statistical audience measurement services are necessary for the operation of the site"_. That's false - it's setting Piwik Pro analytics and the 180-day A/B bucket here, neither are essential for operation (I'd say?).

Actually, even if you decline consent, cookies will still land. Which violates most EU privacy laws. This is easily provable with a curl request (from user who has not accepted cookies, and has declined tracking) like: `curl -sI -H 'DNT: 1' -H 'Sec-GPC: 1' https://www.qwant.com/ | grep -i set-cookie`

---

### Small security issues
Generally all good. From obvious public info, the only issues you'll notice are:
- Their CORS setup is broken. They've put an unanchored allow for `.qwant.com` (which might _look_ correct) but this allows any origin, anywhere containing the word `qwant.com` (like  `www.qwant.com.attacker.net`) to make credentialed requests, because they also have `Access-Control-Allow-Credentials: true` set.
- They've also misspelled permissions policy (they put `permission-policy` instead of `Permissions-Policy` (plural)). So it is non-functional, and all browser permissions (camera, microphone, geoloadtion, payment, USB, etc) are allowed).
- And the CSP on the actual search page is unusually permissive: `script-src 'self' 'unsafe-inline' 'unsafe-eval' data: *.qwant.com`, so any script injected there will execute. But the auth page is much better CSP.
- They put bot protection on the wrong layer. The `/kratos/self-service/login/{api,browser}` endpoint totally bypasses DataDome, and that's the one which you'd use to crack an account
- DNS all checks out. Nothing unusual but the TXT records expose some stack info: Cursor, Anthropic/Claude (x2), Mistral, Notion, Slack, Riot, Apple, Microsoft, Google (x6). They use Google Workspace but haven't deployed MTA-STS (so an internal attacker can strip STARTTLS on inbound mail), but this is fairly normal.

---

### Few more minor policy issues

- DNT gets read, sent but never respected. Testing shows setting `DNT: 1` and `Sec-GPC: 1` in the browser settings, correctly appends `"doNotTrack":true`. However Piwik beacon is still present, Qwant is literally recording my "do not track" request in the tracking endpoint, giving it an extra data point to track you on. It's making users more identifiable, not less!
- Even just reading the privacy policy gets you tracked. It keeps pinging the beacon to record how long you're reading for
- There's a gap for Qwant Junior. Privacy policy says it's not covered, but Qwant junior has full-on adult tracking, despite it being under-16s
- There's some undisclosed infra and third-parties still missing from the privacy policy. Like Elastic APM RUM, Revive ad server, survey / feedback APIs, etc

**Comment by lissy93** [comment]
I think it's time to remove Qwant. Would you agree? PR ready in #738

The problem with the search engines category is, that there's really just no good privacy options. Building/running a search engine seems to just be really hard and expensive. Anything semi-good is just a Bing wrapper, and nothing is open source or verifiable. 

I'll look into the other stuff listed in this category next. 
But I know that if we dig for issues, we'll always find them. And really I don't want to remove everything or only list the super hardcore things here, since it ends up just pushing people towards staying on Google, which is the worst possible outcome.

**Comment by ltguillaume** [comment]
I wouldn't have thought it would be this bad 🙈

I was already typing a reply when you sent the message above. And I see you already created a PR to remove Qwant. I'd certainly agree with that. Even if they respond and make changes, it's clear from the privacy policy alone that it's not a priority.

Well, good thing we weren't adding Qwant to the LibreWolf defaults then 😅 That said, I would still hope they take your findings seriously, otherwise it would once again be painfully clear that to many, "privacy" is a marketing tool and nothing more.

And I fear this isn't much different for Startpage, for that matter. I see uBlock0 already blocks `||startpage.com/clev`, `||startpage.com/perf` and `||startpage.com/sp/atq/*/getconfig/sodar`, for example. In the past there also was a difference between eu.startpage.com and startpage.com with regard to what servers were used and I believe also where "anonymized stats" were sent to, but I have no clue if that's still the case.

Thank you so much for looking into this (again), it's sad to see Qwant's priorities seem to be just for show, that they simply violate the GDPR and fail to protect children from being tracked.

>I don't want to remove everything or only list the super hardcore things here, since it ends up just pushing people towards staying on Google, which is the worst possible outcome.

Very true. There are many initiatives, but the small ones are just meta search engines that will eventually be blocked (recently Whoogle, for example), and big ones need money and - apparently - this can't be done without doing concessions (or blatantly lying).

**Comment by lissy93** [comment]
Qwant replied same-day. I'm grateful they took the time to reply, and are going to look into this, and make corrections if needed. 

<img width="500" alt="Image" src="https://github.com/user-attachments/assets/299e25d6-c73f-40c4-8e3e-e2cb077f75b7" />

Up next, Startpage...

**Comment by lissy93** [comment]
## Startpage

As @ltguillaume mentioned, there is a fair bit of ~~tracking~~ analytics 😔 
From the frontend requests only, these do at least all appear to be first-party, and line up with the privacy policy. 
But ofc no way to tell what's happening to all that info once it's server-side


The good:
- 🟢 No cookies on homepage, just the short-lived bot protection cookie (self-hosted Anubis instance)
- 🟢 No active third-party trackers, analytics or services (that I can see)
- 🟢 They have a nice "Anonymous View" feature, to preview search results via their proxy
- 🟢 No obvious signs of browser fingerprinting, beyond the anti-abuse features and ad-block check

Issues: 
- 🟠 Uses Google (and  Bing + Airfind) for search results, so still censored/controlled. But their udog proxy does strip all IP and user data before querying Google
- 🟠 There is a LOT of first-party analytics/monitoring: `/clev` is click events, `/sp` is their ad ping backs, and `/perf` being the RUM beacon.
- 🟠 Heavily relies on AWS (Cloudfront + S3) and other US-based big tech providers
- 🟠 Mobile app has Sentry and Amplitude. Both disclosed in privacy policy, opt-in
- 🟠 Not open source
- 🟠 No security policy. I cannot find anywhere how to report security issue
- 🟠 Privacy policy doesn't mention anything about the AI search results / which models see your data
- 🟠 They do use your IP for determining location (`req_loc` stored lat/lng). Probably necessary, but makes "we don't record your IP" sound slightly disingenuous 
- 🟠 Privacy policy does not disclose partners, beyond just Google and "various sources". Seems they might use ads from Bing or Airfind too, but hard to confirm this
- 🟠 The privacy policy never mentions where the data lives, anything about ownership, or how Google fits in (results come from Google, so relevant)
- 🟠 Weak CSP and no permissions policy, minor on it's own, but important in regards to the hijacked subdomain (see below)

Bigger issues:
- 🔴 Some ad-block detection going on. A hidden form field within a fake `<div class="a-bg-result AdSense Ads_head...` element, checks if user has ad-block, and appends `abd` (detected) / `abe` (enabled) directly into each search query URL (like `&abd=1`), making you more identifiable. And this is sent to /clev endpoint too.
- 🔴 AND while currently disabled/inactive, it looks like they do have some third-party ad-tech / ad-block-bypass stuff shipped to the client. I can see `ad_admiral:"spicy"` for Admuiral, as well as `cheq_settings:{disable_cheq:true}` for identifying users between bots. Both appear disabled but seem fully setup, neither disclosed in privacy policy.

Side notes:
- There could be things I've missed. Or something i've misinterpreted / got wrong
- The biggest surface is going to be server-side. We have no way to check this.
- Categorization / colored circles just my opinion
- I think these write-ups can have a tendency to seem overly negative. If you look into anything, there's almost always issues like these

### Security Issue

As for security, the only weird thing I noticed is the subdomain take over attack from: https://pluto.startpage.com. 
Probably caused by a dangling IP (`50.116.57.7`) which is now owned by someone else (a Linode VM hosting some fake school thing).

This is an issue because:
- Startpage's lax CSP means that any .startpage.com URL will happily execute any bad scripts hosted on that pluto site
- The bad site can read and write any `SameSite=None` cookies on startpage domain (like harvest secrets from employee Auth0 portal)
- Right now, it seems to just be serving an SEO doorway spam. WITH lots of ads and trackers
- But an attacker could easily make a fake login page or something much worse there, and it would appear totally legitimate

**Comment by ltguillaume** [comment]
Thanks again for all this work, much appreciated!

>There is a LOT of first-party tracking: /clev is click events, /sp is their ad ping backs, and /perf being the RUM beacon.

While I am certainly not as knowledgeable as you are in this, I think it would be good to note there the distinction here between "tracking" and "analytics", right? What is mentioned here seems to be primarily analytics.

Any idea what the query `sc` parameter is?

>Heavily relies on AWS (Cloudfront + S3) and other US-based big tech providers

I wonder how much the difference between https://startpage.com and https://eu.startpage.com is still present. There's a specific setting for which servers to use at the bottom of https://eu.startpage.com/do/settings (but for some reason it's still set to `Fastest / Closest` if you explicitly go there via the `eu` subdomain).

>No security policy. I cannot find anywhere how to report security issue

🤯 Wow, this is bad, especially considering you already found the subdomain hijack. What's next, a... Reddit post 😅 Or perhaps their Insiders program has a proper way for reporting this (https://redlib-2.privadency.com/r/StartpageSearch/wiki/index#wiki_interested_in_helping_shape_startpage.3F).

>Privacy policy doesn't mention anything about the AI search results / which models see your data

I'm not sure what AI search results you mean: I don't see them. Perhaps this is a difference between `eu.startpage` and `startpage` then? I do see they have some app called Vanish AI now.

There's also the issue of Startpage presenting people with captcha's. I primarily see them when I do a query that includes double quotes, but other people see them without being able to determine a specific reason. VPN is certainly a factor, though.

**Comment by lissy93** [comment]
> I think there would be distinction here between "tracking" and "analytics", right, and what is mentioned here seems to be primarily analytics.

Yeah, you're right. Analytics, not tracking - I'll edit 👍

> I wonder how much the difference between https://startpage.com/ and https://eu.startpage.com/

I think eu page is basically the same. Weirdly, when set to "Fastest/closest" for me it kept choosing the US server, until I explicitly set it to EU. Even still, it's AWS eitherway, so technically US could still compel them (under CLOUD act), but it's not likely

> There's also the issue of Startpage presenting people with captcha's

I'm really surprised I didn't get captcha'd that much, when trying so many weird requests 😅
But credit to them for using Anubis + privacy-respecting captcha. The big anti-bot companies (like DataDome that Qwant uses) are just horrible for privacy

**Comment by ltguillaume** [comment]
> I think eu page is basically the same. Weirdly, when set to "Fastest/closest" for me it kept choosing the US server, until I explicitly set it to EU. Even still, it's AWS eitherway, so technically US could still compel them (under CLOUD act), but it's not likely

Yeah, I got all CDN and `adsense/search/async-ads.js` (why wouldn't this be blocked by uBlock0?) from Poland (67.63.60.231) and the rest form from Germany (3.161.82.24), but that's not a guarantee that it's because of `eu.startpage`. I guess I'll need to do some VPN testing.

> I'm really surprised I didn't get captcha'd that much, when trying so many weird requests 😅

I guess it'll also depend on which (if any) VPN you're using?

>But credit to them for using Anubis + privacy-respecting captcha. The big anti-bot companies (like DataDome that Qwant uses) are just horrible for privacy

That's definitely true.

@lissy93 
Do you plan on informing Startpage about your findings?

**Comment by lissy93** [comment]
Yeah, well I sent them a message through their generic support channel. They never replied, but `https://pluto.startpage.com/` has been taken down, so I guess they did get it

**Comment by ltguillaume** [comment]
Well, there's that at least then... Maybe they'll still reply.

**Comment by arnon-shub** [comment]
Late here, but three things I don't think are in the thread.

The CNIL already ruled on Qwant on 11 February 2025: the data sent to Microsoft was pseudonymised, not anonymised, contrary to how Qwant described it, and they were told to align the language versions of the policy. The French version now shows 05/02/2026 while English is 20/05/2024 and German 12/06/2024. Same misdescription, same unaligned policies, eighteen months on.

Qwant Junior is the sharpest finding. The ICO Children's code, standard 12, requires profiling off by default and says consent is the only viable basis for behavioural advertising, and it explicitly covers search engines. GDPR Article 8 sets the age at 15 in France, so most of the target audience cannot consent at all.

On Startpage, Chouaki et al. (ACM IMC 2023) measured the server side you can't see: Google identifies the user in 100% of ad clicks there, and advertisers get identifiers in 92%, against 68% for DuckDuckGo. Worth adding that System1's FY2025 filing carried a going-concern warning, 67% of revenue came from Google, and on 5 August they launched IntentStream, an audience data product spanning their owned properties. Whether Startpage feeds it is a better question for them than the subdomain.
